adcm_aio_client/core/objects/cm.py

```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
 class BundlesNode(PaginatedAccessor[Bundle, None]):
     class_type = Bundle
 
+    async def _download_external_bundle(self: Self, url: str) -> bytes:
+        try:
+            urlopen(url)  # noqa S310
+            async with httpx.AsyncClient() as client:
+                response = await client.get(url)
+                return response.content
+        except ValueError as err:
+            raise OperationError from err
+
+    async def create(self: Self, bundle_loc: str, accept_license: bool | None = None) -> Bundle:
+        if not Path(bundle_loc).exists():
+            file_content = await self._download_external_bundle(bundle_loc)
+            files = {"file": file_content}
+        else:
+            files = {"file": Path(bundle_loc).read_bytes()}
+
+        response = await self._requester.post("bundles", data=files)
+
+        if accept_license:
+            (await self.get()).license.accept()
+
```

Comment:
1. No need in defaulting to `None` for accept license, just `accept_license: bool = False`
2. Requirements defines signature for source as `source: UrlPath | Path`, where `Path` is `pathlib.Path` object (so you can check intentions by checking `isinstance(source, Path)`) and `UrlPath` is either string or our custom string wrapper (probably our custom type is better for now, so you can check it's type too). See that argument is named differently in requirements

Reply:
changed